cumulus_library_catalog/catalog/icd10_annotated_counts.py

```diff
@@ -1,61 +1,81 @@
 import cumulus_library
-from cumulus_library import study_manifest
-
+from cumulus_library import errors, study_manifest
 
 
 class IcdCountsBuilder(cumulus_library.CountsBuilder):
     display_text = "Creating ICD10 counts..."
-    def get_icd10_diagnosis_query(self, min_subject: int | None = None):
+
+    def get_icd10_diagnosis_query(self, table_name, depth, min_subject: int | None = None):
         """Convenience function for getting icd10 table
 
         Primarily exists for overriding min_subject for small bins in
         test data.
         """
+        potential_cols = [
+            (("chapter_code", None),("chapter_str", None)),
+            (("block_code", None),("block_str", None)),
+            (("category_code", None),("category_str", None)),
+            (("subcategory_1_code", None),("subcategory_1_str", None)),
+            (("subcategory_2_code", None),("subcategory_2_str", None)),
+            (("subcategory_3_code", None),("subcategory_3_str", None)),
+            (("extension_code", None),("extension_str", None)),
+        ]
+        if depth > len(potential_cols) | depth <= 0:
+            raise errors.CountsBuilderError(
+                f"Requested ICD10 depth expected to be between 1 and {len(potential_cols) / 2}"
+            )
+        annotation_cols = []
+        # if not at the tree root, get the parent tier for annotation
+        if depth != 0:
+            annotation_cols.append(potential_cols[(depth - 1)][0])
+            annotation_cols.append(potential_cols[(depth - 1)][1])
```

Comment:
You could switch these two lines to just one: `annotation_cols.extend(potential_cols[(depth - 1)])`